rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java

```diff
@@ -184,6 +184,7 @@ public static GitProvenance fromProjectDirectory(Path projectDir) {
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         } catch (IllegalArgumentException | GitAPIException e) {
+            e.printStackTrace();
```

Comment:
We might have suppressed it for tests and recipe runs run outside of git repositories, such as those in build plugins which lay out a directory structure with an example gradle/maven build. Can you make sure that for a non-git repository that null is returned from here with no stack trace?

Reply:
Hopefully covered in 2c9a1800553969351940b3c4c5b6a54f5a3d0408; not guarded against null/empty stacktracte, but I suppose that's fine.